sematic/plugins/abstract_publisher.py

```diff
@@ -0,0 +1,32 @@
+# Standard Library
+import abc
+from typing import Any, List, Type, cast
+
+# Sematic
+from sematic.abstract_plugin import AbstractPlugin, PluginScope
+from sematic.config.settings import get_active_plugins
+
+
+class AbstractPublisher(abc.ABC):
+    """
+    Abstract base class to represent an event external publisher.
+    """
+
+    @abc.abstractmethod
+    def publish(self, event: Any) -> None:
+        pass
+
+
+def get_publishing_plugins(
+    default: List[Type[AbstractPlugin]],
+) -> List[Type[AbstractPublisher]]:
+    """
+    Return all configured "PUBLISH" scope plugins.
+    """
+    publisher_plugins = get_active_plugins(scope=PluginScope.PUBLISH, default=default)
+
+    publisher_classes = [
+        cast(Type[AbstractPublisher], plugin) for plugin in publisher_plugins
+    ]
+
+    return publisher_classes
```

Comment:
O how I wish we could centralize this casting logic for all plug-ins types. Unfortunately, we cannot write a function that takes in `type_=Type[AbstractPublisher]` and does `cast(type_, plugin)`.

Reply:
This is what generics are for!

```
T = TypeVar(T, bound=AbstractPlugin)

def get_active_plugins(plugin_class: Type[T], default: Optional[List[T]]=None) -> List[T]:
    scope = plugin_class.scope
    # ...

# MyPy should accept this
publisher_plugins List[AbstractPublisher] = get_active_plugins(plugin_class=AbstractPublisher)
```

Reply:
Ah yes, I did try with generic, but my mistake case to try to do the casting manually. Maybe this will work. Although the output should be `List[Type[T]]`.